Storage/SQL/SQLiteRemoteClientsAndTabs.swift

```diff
@@ -347,6 +347,60 @@ open class SQLiteRemoteClientsAndTabs: RemoteClientsAndTabs {
     }
 }
 
+extension SQLiteRemoteClientsAndTabs: RemoteDevices {
+    open func replaceRemoteDevices(_ remoteDevices: [RemoteDevice]) -> Success {
+        // Drop corrupted records and our own record too.
+        let remoteDevices = remoteDevices.filter { $0.id != nil && $0.type != nil && !$0.isCurrentDevice }
+
+        let deferred = Success()
+        var err: NSError?
+        let resultError = self.db.transaction(&err) { (conn, err: inout NSError?) in
+            func change(_ sql: String, args: Args?=nil) -> Bool {
+                if let e = conn.executeChange(sql, withArgs: args) {
+                    err = e
+                    deferred.fillIfUnfilled(Maybe(failure: DatabaseError(err: e)))
+                    return false
+                }
+                return true
+            }
+
+            _ = change("DELETE FROM \(TableRemoteDevices)")
+
+            if err != nil {
+                return false
+            }
+
+            let now = Date.now()
```

Comment:
File a follow-up: use the server timestamp of the fetch for this? (Perhaps by adding the dates to `RemoteDevice`.)